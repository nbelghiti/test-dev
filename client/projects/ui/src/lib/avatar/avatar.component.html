<span
	class="c-avatar"
	[ngClass]="{
		'is-compact': compact
	}"
>
	<abbr [title]="name">{{initials}}</abbr>
</span>
