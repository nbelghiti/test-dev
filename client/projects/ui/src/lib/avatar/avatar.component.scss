// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$avatar-size-base: 4.4rem;
$avatar-size-compact: 3.2rem;

// =============================================================================
// :: Avatar
// =============================================================================
.c-avatar {
	display: flex;
	justify-content: center;
	align-items: center;
	width: $avatar-size-base;
	min-width: $avatar-size-base;
	height: $avatar-size-base;

	font-size: $font-size-1x-large;
	font-weight: $font-weight-bold;
	line-height: $line-height-base;
	text-transform: uppercase;
	text-align: center;

	border-radius: 50%;
	background-color: $background-color-grey-96;

	cursor: default;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-avatar {

	&.is-compact {
		width: $avatar-size-compact;
		min-width: $avatar-size-compact;
		height: $avatar-size-compact;

		font-size: $font-size-base;
	}
}
