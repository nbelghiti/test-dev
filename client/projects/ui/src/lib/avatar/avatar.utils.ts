export const getInitials = (name: string): string => {
	const names = name.split(' ');
	let initials = names[0].slice(0, 1);

	if (names.length > 1) {
		initials += names[names.length - 1].slice(0, 1);
	}

	return initials.toUpperCase();
};
