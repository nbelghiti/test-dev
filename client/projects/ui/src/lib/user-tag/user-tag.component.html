<div
	class="c-user-tag"
	[class.c-user-tag--fluid]="fluid"
	[title]="user?.name"
>
	<div class="c-user-tag__content">
		<ui-user
			[user]="user"
		>
		</ui-user>
	</div>

	<div
		*ngIf="icon && !isClickedUsed"
		class="c-user-tag__icon"
	>
		<ui-icon
			[name]="icon"
			[size]="IconSize.tiny"
		>
		</ui-icon>
	</div>

	<button
		*ngIf="isClickedUsed"
		type="button"
		(click)="clicked.emit(user)"
		class="c-user-tag__button"
	>
		<ui-icon
			[name]="IconName.times"
			[size]="IconSize.tiny"
		>
		</ui-icon>
	</button>
</div>
