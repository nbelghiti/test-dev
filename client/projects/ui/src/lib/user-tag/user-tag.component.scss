// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: User tag
// =============================================================================
.c-user-tag {
	display: inline-flex;
	align-items: center;

	background-color: $background-color-grey-96;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-user-tag__content {
	padding: 0.4rem 1rem;
}

.c-user-tag__icon {
	padding: 0 1rem;

	color: $text-color-base;
}

.c-user-tag__button {
	align-self: stretch;
	padding: 0 1rem;

	color: $text-color-grey-46;

	transition:
		color $animation-base,
		background-color $animation-base;

	&:hover {
		background-color: $background-color-grey-86;
		color: $text-color-base;
	}
}

// =============================================================================
// Modifier
// =============================================================================
.c-user-tag--fluid {
	width: 100%;
}
