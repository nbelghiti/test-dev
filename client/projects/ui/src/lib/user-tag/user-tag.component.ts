import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';

import { IconName, IconSize } from '../icon/icon.types';
import { IUser } from '../user/user.types';

@Component({
	selector: 'ui-user-tag',
	templateUrl: './user-tag.component.html',
	styleUrls: ['./user-tag.component.scss']
})

export class UserTagComponent implements OnInit {
	@Input() public user: IUser;
	@Input() public icon: IconName;
	@Input() public fluid: boolean = false;
	@Output() public clicked: EventEmitter<IUser> = new EventEmitter<IUser>();

	public isClickedUsed = false;
	public IconName: typeof IconName = IconName;
	public IconSize: typeof IconSize = IconSize;

	ngOnInit() {
		this.isClickedUsed = this.clicked.observers.length > 0;
	}
}
