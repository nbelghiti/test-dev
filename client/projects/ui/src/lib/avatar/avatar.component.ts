import { Component, Input } from '@angular/core';

import { getInitials } from './avatar.utils';

@Component({
	selector: 'ui-avatar',
	templateUrl: './avatar.component.html',
	styleUrls: ['./avatar.component.scss']
})

export class AvatarComponent {
	@Input() public compact: boolean = false;

	public initials = '';

	@Input() set name(name: string) {
		if (name) {
			this.initials = getInitials(name);
		}
	}
}
