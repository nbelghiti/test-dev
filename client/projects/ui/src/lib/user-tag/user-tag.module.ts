import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { IconModule } from '../icon/icon.module';
import { UserModule } from '../user/user.module';

import { UserTagComponent } from './user-tag.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
		UserModule,
	],
	declarations: [
		UserTagComponent,
	],
	exports: [
		UserTagComponent,
	],
})
export class UserTagModule {}
