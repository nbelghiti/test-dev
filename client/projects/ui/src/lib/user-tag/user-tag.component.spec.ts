import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { TranslateModule, TranslateService } from '@ngx-translate/core';

import { AvatarComponent } from '../avatar/avatar.component';
import { IconComponent } from '../icon/icon.component';
import { UserComponent } from '../user/user.component';

import { UserTagComponent } from './user-tag.component';

const mockUser = {
	name: 'John',
	role: '',
	type: 'engineer',
	myself: false,
};

describe('UserTagComponent', () => {
	let component: UserTagComponent;
	let fixture: ComponentFixture<UserTagComponent>;
	let translate: TranslateService;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				TranslateModule.forRoot(),
			],
			declarations: [
				IconComponent,
				UserComponent,
				UserTagComponent,
				AvatarComponent
			],
			providers: [
				TranslateService,
			]
		})
			.compileComponents();

		// Get the providers from the TestBed
		translate = TestBed.get(TranslateService);
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(UserTagComponent);
		component = fixture.componentInstance;
		component.user = mockUser;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
