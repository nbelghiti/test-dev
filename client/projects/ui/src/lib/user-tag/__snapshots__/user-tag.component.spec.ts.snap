// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTagComponent Should render correctly 1`] = `
<ui-user-tag
  IconName={[Function Object]}
  IconSize={[Function Object]}
  clicked={[Function EventEmitter]}
  fluid="false"
  isClickedUsed="false"
  user={[Function Object]}
>
  <div
    class="c-user-tag"
    title="John"
  >
    <div
      class="c-user-tag__content"
    >
      <ui-user
        ng-reflect-user="[object Object]"
      >
        
        
        <button
          class="c-user"
          ng-reflect-klass="c-user"
          ng-reflect-ng-class="[object Object]"
          title="John"
        >
          
          <ui-avatar
            ng-reflect-compact="true"
            ng-reflect-name="John"
          >
            <span
              class="c-avatar is-compact"
              ng-reflect-klass="c-avatar"
              ng-reflect-ng-class="[object Object]"
            >
              <abbr
                title="undefined"
              >
                J
              </abbr>
            </span>
          </ui-avatar>
          <div
            class="c-user__content"
          >
            <span
              class="c-user__name"
            >
               John
              
            </span>
            <span
              class="c-user__role"
            >
               TEAM_MEMBER_ROLE__engineer 
            </span>
          </div>
        </button>
        
        
      </ui-user>
    </div>
    
    
  </div>
</ui-user-tag>
`;
